ctapipe/image/cleaning.py

```diff
@@ -111,6 +113,67 @@ def tailcuts_clean(
         )
 
 
+def time_clustering(
+    geom,
+    image,
+    time,
+    pedestal=1.6,
+    n_noise=2.5,
+    minpts=5,
+    eps=1.0,
+    t_scale=4.0,
+    d_scale=0.25,
+):
+    """
+    Clean an image by selecting pixels which pass a time clustering algorithm using DBSCAN.
+    Previously used for HESS [timecleaning]_.
+    Parameters
+    ----------
+    geom: `ctapipe.instrument.CameraGeometry`
+        Camera geometry information
+    image: array
+        pixel values
+    time: array
+        pixel values
+    pedestal: array
+        pedestal width for each pixel. This is just one number for now.
+    n_noise: array
+        Noise cut
+    minpts: int
+        Minimum number of points to consider a cluster
+    eps: float
+        Distance in dbscan
+    t_scale: float
+        Time scale in ns
+    d_scale: float
+        Space scale in m
+
+    Returns
+    -------
+    A boolean mask of *clean* pixels.
+    """
+    precut_mask = image > n_noise * pedestal
+
+    arr = np.zeros(len(image), dtype=float)
+    arr[~precut_mask] = -1
+
+    pix_x = geom.pix_x.value[precut_mask] / d_scale
+    pix_y = geom.pix_y.value[precut_mask] / d_scale
+
+    X = np.column_stack((time[precut_mask] / t_scale, pix_x, pix_y))
+
+    db = DBSCAN(eps=eps, min_samples=minpts).fit(X)
```

Comment:
slightly shorter: `labels = DBSCAN(...).fit_predict(X)`.

Reply:
Shouldn't this also use `sample_weights=image`? Otherwise you are using image only for the initial cut

Reply:
I tried `sample_weight`  before. I used this weighting:
$\frac{1}{1+exp\left(-\frac{image+s}{d}\right)}$
It introduces two new parameters but I can add weights as an option

Reply:
I have been thinking about this and I think we could introduce a fourth dimension in the cluster, which is 1/SNR, SNR is the signal-to-noise ratio, image/pedestal. Using just image does not work as the algorithm finds clusters with similar data points but if you plot 1/image, we find clusters:
![image](https://github.com/cta-observatory/ctapipe/assets/53437218/32b5be40-38ee-415f-86b9-282c08569672)

Reply:
It does not seem to make any difference though